src/linter/types.go

```diff
@@ -149,6 +153,34 @@ func (conv *phpdocTypeConverter) mapShapeType(params []phpdoc.TypeExpr) []meta.T
 	return []meta.Type{{Elem: shape.name}}
 }
 
+func (conv *phpdocTypeConverter) mapTupleType(params []phpdoc.TypeExpr) []meta.Type {
+	types := make([]phpdoc.TypeExpr, 0, len(params))
+	for i, p := range params {
+		if p.Value == "*" || p.Value == "..." {
+			continue
+		}
+		if p.Kind != phpdoc.ExprName && p.Kind != phpdoc.ExprGeneric {
+			conv.warn(fmt.Sprintf("tuple param #%d: want type, found %s", i+1, p.Value))
+			continue
+		}
+
+		args := []phpdoc.TypeExpr{
+			phpdoc.TypeExpr{
+				Kind:  phpdoc.ExprInt,
+				Value: fmt.Sprint(i),
+			}, p}
+
+		typeExpr := phpdoc.TypeExpr{
+			Kind:  phpdoc.ExprKeyVal,
+			Value: fmt.Sprintf("%d:%s", i, p.Value),
```

Comment:
I think `mapShapeType` doesn't need ExprKeyVal value. It only uses Key (args[0]) value to determine which array key shape type should match.

If it's not required, maybe we can avoid initializing it?